Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -105,6 +105,80 @@ GstPad* webkitGstGhostPadFromStaticTemplate(GstStaticPadTemplate* staticPadTempl
     return pad;
 }
 
+#if !GST_CHECK_VERSION(1, 20, 0)
+gchar* webkitGstCodecUtilsCapGetMIMECodec(GstCaps* caps)
+{
+    gchar* mimeCodec = nullptr;
+    GstStructure* capsSt = nullptr;
+    const gchar* mediaType = nullptr;
+
+    g_return_val_if_fail(caps != nullptr, nullptr);
+    g_return_val_if_fail(gst_caps_is_fixed(caps), nullptr);
+
+    capsSt = gst_caps_get_structure(caps, 0);
+    if (capsSt == nullptr) {
+        GST_WARNING("Failed to get structure from caps");
+        goto done;
+    }
+
+    mediaType = gst_structure_get_name(capsSt);
+
+    if (!g_strcmp0(mediaType, "video/x-h264")) {
+        /* avc1.AABBCC
+         *   AA = profile
+         *   BB = constraint set flags
+         *   CC = level
+         */
+        guint8 profile = 0;
+        guint8 flags = 0;
+        guint8 level = 0;
+
+        if (!h264_capsStructure_get_profile_flags_level(capsSt, &profile, &flags, &level)) {
+            GST_DEBUG("h264 caps did not contain 'codec_data', cannot determine detailed codecs info");
+            mimeCodec = g_strdup("avc1");
+        } else
+            mimeCodec = g_strdup_printf("avc1.%02X%02X%02X", profile, flags, level);
+    } else if (!g_strcmp0(mediaType, "video/x-h265")) {
+        if (!hevc_caps_get_mimeCodec(caps, &mimeCodec)) {
```

Comment:
This and a few others are private to codec-utils.c ... I don't think it's worth backporting this TBH.

Reply:
OK, so then it's necessary to replace 'gst_codec_utils_caps_get_mime_codec' for something else.